src/firefly/java/edu/caltech/ipac/firefly/server/query/ExtractFromImage.java

```diff
@@ -57,38 +59,38 @@ public DataGroup fetchDataGroup(TableServerRequest req) throws DataAccessExcepti
                 String wlUnit= req.getParam(ServerParams.WL_UNIT);
                 Map<Integer,String> fluxUnit= makeMapOfUnitsFromParam(req);
                 double[] wlAry= SrvParam.getDoubleAryFromJson(req.getParam(ServerParams.WL_ARY));
-                return FITSTableReader.getCubeZaxisAsTable(pt, wpt, filename, refHduNum, allMatchingHDUs, extractionSize, wlAry,wlUnit,fluxUnit);
+                return FITSExtractToTable.getCubeZaxisAsTable(pt, wpt, filename, refHduNum, allMatchingHDUs,
+                        extractionSize, ct, wlAry,wlUnit,fluxUnit);
             }
             else if (extType.equals("line")) {
                 ImagePt[] ptAry= SrvParam.getImagePtAryFromJson(req.getParam(ServerParams.PTARY));
                 WorldPt[] wptAry= SrvParam.getWorldPtAryFromJson(req.getParam(ServerParams.WPT_ARY));
                 checkPointParams(ptAry, plane, filename, refHduNum);
-                return FITSTableReader.getLineSelectAsTable(ptAry, wptAry, filename, refHduNum, plane, allMatchingHDUs, extractionSize);
+                return FITSExtractToTable.getLineSelectAsTable(ptAry, wptAry, filename, refHduNum, plane, allMatchingHDUs,
+                        extractionSize, ct);
             } else if (extType.equals("points")) {
                 ImagePt[] ptAry= SrvParam.getImagePtAryFromJson(req.getParam(ServerParams.PTARY));
                 WorldPt[] wptAry= SrvParam.getWorldPtAryFromJson(req.getParam(ServerParams.WPT_ARY));
                 checkPointParams(ptAry, plane, filename, refHduNum);
-                return FITSTableReader.getPointsAsTable(ptAry, wptAry, filename, refHduNum, plane, allMatchingHDUs, extractionSize);
+                return FITSExtractToTable.getPointsAsTable(ptAry, wptAry, filename, refHduNum, plane,
+                        allMatchingHDUs, extractionSize, ct);
             }
         } catch (IOException | FitsException e) {
             throw new IllegalArgumentException("Could not make a table from extracted data");
         }
         throw new IllegalArgumentException("extractionType must be z-axis, line, or points");
     }
 
-
     private static Map<Integer,String> makeMapOfUnitsFromParam(TableServerRequest req) {
         String[] sAry= SrvParam.getStringAryFromJson(req.getParam(ServerParams.FLUX_UNIT));
         Map<Integer,String> fluxUnit= new HashMap<>();
-        if (sAry!=null) {
-            for(String entry : sAry) {
-                String[] s= entry.split("=");
-                if (s.length==2) {
-                    try {
-                        int hdu= Integer.parseInt(s[0]);
-                        fluxUnit.put(hdu,s[1]);
-                    } catch (NumberFormatException ignore) { }
-                }
+        if (sAry==null) return fluxUnit;
+        for(String entry : sAry) {
+            String[] s= entry.split("=");
+            if (s.length==2) {
```

Comment:
FYI, If you use split("=", 2), then you can accept '=' as part of the value.  Not sure what the requirement is.